ush/get_obs.py

```diff
@@ -0,0 +1,1004 @@
+        # and 12-18 of the current day) and another for the next day (which
+        # includes the file for accumulations over hours 18-24 of the current
+        # day).  To distinguish between the raw obs files from these two archives,
+        # we create an archive-time dependent raw subdirectory for each possible
+        # archive.
+        #
+        # MRMS:
+        # There is only one archive per day, and it contains all the raw obs
+        # files needed to generate processed obs files for all hours of the
+        # current day.  Thus, we will only ever need this one archive, so there
+        # is no need to include the archive's hour information (there really
+        # isn't any) in the raw subdirectory name.  In addition, the archive's
+        # year, month, and day is the same as that of the obs day's, so it is
+        # already included in the name of the raw base directory.  Sine this is
+        # the only info we need to avoid differnt get_obs tasks clobbering each
+        # other's output obs files, for simplicity we simply do not create a raw
+        # archive subdirectory.
+        #
+        # NDAS:
+        # Same as for CCPA.
+        if obtype == 'CCPA':
+            arcv_subdir_raw = yyyymmddhh_arcv_str
+        elif obtype == 'NOHRSC':
+            arcv_subdir_raw = yyyymmdd_arcv_str
+        elif obtype == 'MRMS':
+            arcv_subdir_raw = ''
+        elif obtype == 'NDAS':
+            arcv_subdir_raw = yyyymmddhh_arcv_str
+
+        # Combine the raw archive base directory with the raw archive subdirectory
+        # name to obtain the full path to the raw archive directory.
+        arcv_dir_raw = os.path.join(basedir_raw, arcv_subdir_raw)
+
+        # Check whether any of the obs retrieval times for the day associated with
+        # this task fall in the time interval spanned by the current archive.  If
+        # so, set the flag (do_retrieve) to retrieve the files in the current
+        # archive.
+        if obtype == 'CCPA':
+            arcv_contents_start = yyyymmddhh_arcv - (num_obs_files_per_arcv - 1)*obs_avail_intvl
+            arcv_contents_end = yyyymmddhh_arcv
+        elif obtype == 'NOHRSC':
+            arcv_contents_start = yyyymmddhh_arcv
+            arcv_contents_end = yyyymmddhh_arcv + (num_obs_files_per_arcv - 1)*obs_avail_intvl
+        elif obtype == 'MRMS':
+            arcv_contents_start = yyyymmddhh_arcv
+            arcv_contents_end = yyyymmddhh_arcv + (num_obs_files_per_arcv - 1)*obs_avail_intvl
+        elif obtype == 'NDAS':
+            arcv_contents_start = yyyymmddhh_arcv - num_obs_files_per_arcv*obs_avail_intvl
+            arcv_contents_end = yyyymmddhh_arcv - obs_avail_intvl
+
+        do_retrieve = False
+        for obs_retrieve_time in obs_retrieve_times_crnt_day:
+            if (obs_retrieve_time >= arcv_contents_start) and \
+               (obs_retrieve_time <= arcv_contents_end):
+              do_retrieve = True
+              break
```

Comment:
Indentation should be 4 spaces per level

```suggestion
                do_retrieve = True
                break
```

Reply:
Fixed.